server/routes/api/user.ts

```diff
@@ -299,3 +300,220 @@ userRoutes.route("/export").get(isAdmin, async (request, response) => {
 		});
 	}
 });
+
+async function removeUserFromAllTeams(user: IUserMongoose): Promise<boolean> {
+	if (!user.teamId) {
+		return true;
+	}
+
+	let currentUserTeam: ITeamMongoose = await Team.findById(user.teamId);
+	if (!currentUserTeam) {
+		return false;
+	}
+
+	if (currentUserTeam.teamLeader.toString() === user._id.toString()) {
+		// If we're team leader, remove everybody from group
+		await Team.findByIdAndRemove(currentUserTeam._id);
+		await User.update({
+			teamId: user.teamId
+		}, {
+			$unset: {
+				teamId: 1
+			}
+		}, {
+			multi: true
+		});
+		return true;
+	}
+	else {
+		await Team.update({
+			_id: user.teamId
+		}, {
+			$pull: {
+				members: user._id
+			}
+		});
+		// Remove us from the current team we're in
+		await User.update({
+			_id: user._id
+		}, {
+			$unset: {
+				teamId: 1
+			}
+		});
+		return true;
+	}
+}
+
+userRoutes.route("/team/create/:teamName").post(isUserOrAdmin, async (request, response) => {
+	/*
+		First, check if there is already a team that has this user as captain
+		If so, just re-assign the user's teamid to that one
+		Else if the user's in a team, take them out of it
+		Else make them a new team
+	 */
+	let user = await User.findById(request.params.id);
+	let decodedTeamName = decodeURI(request.params.teamName);
+
+	let existingTeam = await Team.findOne({ teamName: decodedTeamName });
+
+	if (existingTeam) {
+
+		// Someone else has a team with this name
+		return response.status(400).json({
+			"success": false,
+			"message": `Someone else has a team called ${decodedTeamName}. Please pick a different name.`
```

Comment:
To remain consistent with the rest of the API endpoints, this object (and the others with the `"success": false` convention) should only have the key-value pair `"error": "error message"`